### PR TITLE
fix (systemaddon) #3626 UX improvements to info icon+box

### LINKED ISSUE #3626: Add info link to all sections.
After hacking up some ideas with @bryanbell, we decided we want to:

* have an info icon on all sections
* only show the icon on hover of the section
* add a link to the preferences sidebar from the info box
* only show the top sites edit link on hover of top sites

We want to uplift this for 57.

### PR DESCRIPTION
**Pull request by rlr**
Fixes #3626 

Most of the changes are moving things around/refactoring so that Top Sites could share the code from Sections. I'll make it better in the CollapsibleSections branch.

### CODE REVIEW

**Comment by Mardak** [comment]
Just doing some quick testing: Highlights doesn't seem to get this because it didn't have an info option. I would think we would should it for Highlights using the strings that we would have shown in the prefs pane: "Highlights" "Find your way…"

Also minor UI nit, but the ">" arrow looks a little high.. maybe half a pixel? Double check with @aaronrbenson 
image


@k88hudson can you also take a look as we're looking to uplift this late to beta 57?

@csadilek mentioned the positioning of the popup is not quite right. Was it vertically too low or too far to the right? There were some horizontal positioning tweaks that I assume were by design https://github.com/mozilla/activity-stream/commit/adae0df2891bb497ce6b2b02b12ab98cdebbf582#diff-dd48c994d9ed857586fa08dfde37fa10R53

But I believe when the spacing of the section top bar were adjusted, the `top` needed to be adjusted too?

**Comment by rlr** [comment]
Oops. I had the highlights info option in another branch but forgot to copy it over. Will fix.

I am waiting for @bryanbell and @aaronrbenson to get in so they can play with it on my machine and approve or ask for tweaks.

File: system-addon/lib/SectionsManager.jsm [hunk 1]
```diff
@@ -52,6 +52,10 @@ const BUILT_IN_SECTIONS = {
     title: {id: "header_highlights"},
     maxRows: 3,
     availableContextMenuOptions: ["CheckBookmark", "SaveToPocket", "Separator", "OpenInNewWindow", "OpenInPrivateWindow", "Separator", "BlockUrl", "DeleteUrl"],
+    infoOption: {
```
Comment (rlr) [inline system-addon/lib/SectionsManager.jsm]: hmm, it is here. it isn't showing up? /me tests

Comment (rlr) [reply system-addon/lib/SectionsManager.jsm]: WFM 😕 

screen shot 2017-09-29 at 9 11 09 am

Comment (Mardak) [reply system-addon/lib/SectionsManager.jsm]: Oh oops. I think I ran without `./mach build browser/app` to update the lib files. The content files updated. Doh!

Comment (Mardak) [reply system-addon/lib/SectionsManager.jsm]: Yup WFM now too.

**Comment by as-pine-proxy** [comment]
[Treeherder: e4e9d3b822695b9967d3502a81a47c030db765d2](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=e4e9d3b822695b9967d3502a81a47c030db765d2)

**Pull request merged**